### PR TITLE
feat: Add AI knowledge assistant Node.js demo

### PR DESCRIPTION
**Pull request by gtondello**
Add AI knowledge assistant Node.js demo.

### CODE REVIEW

File: node/ai-knowledge-assistant/controllers/app.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            url: generateAuthUrl(this.userName, this.configCompleteRedirectUrl)
+          }
+        };
+      }
+      // Rethrow unrecognized errors.
+      throw e;
+    }
+
+    // Reply with welcome message.
+    const text = 'Thank you for adding me to this space.'
```
Comment (danielatgoogle) [inline node/ai-knowledge-assistant/controllers/app.js:123]: Suggested welcome message:

Thank you for adding me to this space. I help answer questions based on past conversation in this space. Go ahead and ask me a question!

Comment (gtondello) [reply node/ai-knowledge-assistant/controllers/app.js]: Done.

File: node/ai-knowledge-assistant/env.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * Project environment settings.
+ */
+const env = {
+  // Replace with your GCP project ID.
+  project: 'project-id',
+
+  // Replace with your GCP project location. Used for Vertex AI calls.
+  location: 'us-central1',
+
+  // Reploace with the PubSub topic to receive events.
```
Comment (allisongh) [inline node/ai-knowledge-assistant/env.js:27]: Replace

**Review by allisongh — COMMENTED** [review]
Left some suggested changes to the env.js file.

File: node/ai-knowledge-assistant/env.js (hunk 2 above)
Comment (gtondello) [reply node/ai-knowledge-assistant/env.js]: Done.

File: node/ai-knowledge-assistant/env.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  // The topic must be in the same GCP project as the Chat app.
+  topic: 'projects/project-id/topics/events-api',
```
[the remark this answers is not in the document]
Comment (gtondello) [reply node/ai-knowledge-assistant/env.js]: Done.

**Review by jpoehnelt — APPROVED** [review]

**Pull request merged**